host/hexdump.c

```diff
@@ -3,9 +3,43 @@
 
 #include <openenclave/host.h>
 #include <openenclave/internal/hexdump.h>
+#include <openenclave/internal/trace.h>
 #include <openenclave/internal/utils.h>
 #include <stdio.h>
 
+void oe_hex_dump_trace(oe_log_level_t level, const void* data_, size_t size)
+{
+    size_t i, block;
+    const unsigned char* data = (const unsigned char*)data_;
+    char* message = NULL;
+
+    if (!data || !size)
+        return;
+
+    if (!(message = malloc(OE_LOG_MESSAGE_LEN_MAX)))
+        return;
+
+    while (size != 0)
```

Comment:
Can't you simplify the code here? Given that you have size already, you can just
- allocate a buffer with size * 2
- `sprintf` against the buffer in a loop using %02x.
- Print the buffer with `OE_TRACE_INFO`

Reply:
Only data upto OE_LOG_MESSAGE_LEN_MAX is logged. Hence the standard size of message buffer. We need the while loop to be able to dump data > max length. host/traceh.c & enclave/core/tracee.c  truncate logs of length greater than OE_LOG_MESSAGE_LEN_MAX.